src/tests/integration/monitor-tabs.test.ts

```diff
@@ -0,0 +1,200 @@
+import { expect } from 'chai'
+import { createMocks } from './mocks'
+import { runBackground } from './steps/run-background'
+
+describe.only('monitor tabs', () => {
+  const mocks = createMocks()
+
+  runBackground(mocks, { alreadyAuthenticated: true })
+
+  describe('chrome.tabs.onCreated', () => {
+    it("adds the newly created tab to the store's state", () => {
+      expect(mocks.chrome.tabs.onCreated.addListener).to.have.callCount(1)
+      const [callback] = mocks.chrome.tabs.onCreated.addListener.firstCall.args
+      callback({
+        id: 18,
+        windowId: 1,
+        active: false,
+        url: 'https://new.com/abc',
+      })
+      const state = mocks.getState()
+      const tab = state.tabs.get(18)!
+      expect(tab.id).to.equal(18)
+      expect(tab.windowId).to.equal(1)
+      expect(tab.domain).to.equal('new.com')
+      expect(tab.url).to.equal('https://new.com/abc')
+    })
+  })
+
+  describe('chrome.tabs.onRemoved', () => {
+    it("removes a tab from the store's state", () => {
+      const [callbackOnRemove] = mocks.chrome.tabs.onRemoved.addListener.firstCall.args
+
+      const tabId = 11
+      const removeInfo = {
+        id: 11,
+        windowId: 1,
+        active: false,
+        url: 'https://foo.com/abc',
+      }
+      callbackOnRemove(tabId, removeInfo)
+
+      const state = mocks.getState()
+      const tabs = state.tabs
+
+      expect(tabs.get(tabId)).to.equal(undefined)
+    })
+  })
+
+  describe('chrome.tabs.onUpdated', () => {
+    it("updates a tab change to store's state", () => {
+      const [callbackOnUpdated] = mocks.chrome.tabs.onUpdated.addListener.firstCall.args
+
+      const tabId = 16
+      const changeInfo = {
+        id: 16,
+        windowId: 3,
+        active: false,
```

Comment:
We only care about url changes, so let's simplify this.

```typescript
const changeInfo = { id: 16, url: 'https://changeinfo.com/abc' }
callback(changeInfo.id, changeInfo)
```

Reply:
Resolved. Thanks!